python_src/src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -364,6 +367,110 @@ def update_directions(db_manager: DatabaseManager) -> None:
     process_logger.log_complete()
 
 
+def update_canonical_stop_sequence(db_manager: DatabaseManager) -> None:
+    """
+    Update canonical_stop_sequence from static_route_patterns
+    """
+    select_update_params = sa.select(
+        TempEventCompare.service_date,
+        TempEventCompare.static_version_key,
+    ).distinct()
+
+    for result in db_manager.select_as_list(select_update_params):
+        service_date = result["service_date"]
+        version_key = result["static_version_key"]
+
+        static_sub = (
```

Comment:
I'm confused why we need the `StaticRoutePatterns` here to get the `stop_sequence` out of the `StaticStopTimes` since it seems like we're getting `StaticStopTimes.stop_sequence` by matching on `trip_id`, which is ultimately coming from the `StaticTrips` table?

Reply:
`StaticRoutePatterns` contains a list of trips to be considered "typical" for each route `route_pattern_typicality=1`. The end result being `representative_trip_id` that is joined to `StaticTrips` to get our `branch_route_id/trunk_route_id` designations that we do ourselves. 

These "typical" trips are joined to `StaticStopTimes` to get `stop_id/stop_sequence` values for each stop in each "typical" trip. 

Finally, the `stop_id` values pulled from `StaticStopTimes` are joined to `StaticStops` to get `parent_station` values we can join to our `vehicle_events` table. 

It's pretty convoluted, but it's the most direct path that I was able to identify.